File: src/RashePharma.Domain/Entities/Quotation.cs
namespace RashePharma.Domain.Entities;

public class Quotation
{
    public int Id { get; set; }

    public int EnquiryId { get; set; }

    public int? UserId { get; set; }

    public string QuoteNumber { get; set; } = string.Empty;

    public decimal TotalAmount { get; set; }

    public string Currency { get; set; } = "USD";

    public string Status { get; set; } = "Draft";

    public DateTime ValidUntil { get; set; }

    public string? Notes { get; set; }

    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public DateTime? UpdatedAt { get; set; }

    // Navigation Properties
    public Enquiry Enquiry { get; set; } = null!;

    public User? User { get; set; }

    public ICollection<QuotationItem> Items { get; set; }
        = new List<QuotationItem>();
}

File: frontend/types/quotation.ts
export interface CreateQuotationItem {
  productVariantId: number;
  quantity: number;
  unitPrice: number;
}

export interface CreateQuotation {
  enquiryId: number;
  currency: string;
  validUntil: string;
  notes?: string | null;
  items: CreateQuotationItem[];
}

export interface QuotationItem {
  id: number;
  productVariantId: number;
  productName: string;
  strength: string | null;
  packSize: string | null;
  quantity: number;
  unitPrice: number;
  totalPrice: number;
}

export interface QuotationDetails {
  id: number;
  quoteNumber: string;
  enquiryId: number;
  totalAmount: number;
  currency: string;
  status: string;
  validUntil: string;
  notes: string | null;
  createdAt: string;
  updatedAt: string | null;
  items: QuotationItem[];
}

export interface QuotationList {
  id: number;
  quoteNumber: string;
  enquiryId: number;
  totalAmount: number;
  currency: string;
  status: string;
  validUntil: string;
  createdAt: string;
}